libraries/provision/generate_pools_json_from_aws.py

```diff
@@ -74,10 +77,78 @@ def get_public_dns_names_cloudformation_stack(stackname):
     # wait until all ec2 instances are listening on port 22
     wait_until_sshd_port_listening(instances_for_stack)
 
+    return instances_for_stack
+
+
+def get_public_dns_names_cloudformation_stack(stackname):
+
+    """
+    Returns the public DNS names of all EC2 instances in the stack
+    """
+
+    instances_for_stack = get_running_instances_for_cloudformation_stack(stackname)
+
     # get public_dns_name for all instances
     return get_public_dns_names(instances_for_stack)
 
 
+def ip_to_ansible_group_for_cloudformation_stack(stackname):
+
+    """
+    Generate a dictionary like:
+
+    "ip_to_node_type": {
+       "s61702cnt72.sc.couchbase.com": "couchbase_servers",
+       "s61703cnt72.sc.couchbase.com": "couchbase_servers",
+       "s61704cnt72.sc.couchbase.com": "couchbase_servers",
+       "s61705cnt72.sc.couchbase.com": "sync_gateways",
+       ....
+    }
+    """
+
+    instances_for_stack = get_running_instances_for_cloudformation_stack(stackname)
+
+    ip_to_ansible_group = {}
+    for instance in instances_for_stack:
+        ansible_group = get_ansible_group_for_instance(instance)
+        ip_to_ansible_group[instance.public_dns_name] = ansible_group
+
+    return ip_to_ansible_group
+
+
+def get_ansible_group_for_instance(instance):
+
+    """
+    Given an ec2 instance:
+
+    1. Look for the "type" tag, which will be something like "couchbaserver", which is set in cloudformation template
+    2. Translate that to the expected "node_type" that corresponds to ansible group
+
+    NOTE regarding sg_accels:  they are tagged with type=syncgateway, but they also have a CacheType="writer" tag
+    that we can use to differentiate them from sync gateways
+
+    """
+
+    instance_type_to_ansible_group = {
```

Comment:
Yes good point, they are assigned here:

https://github.com/couchbaselabs/mobile-testkit/blob/master/libraries/provision/cloudformation_template.py#L164

If we sync up the tags to avoid this mapping, that might be better to do in a separate issue (or as part of https://github.com/couchbaselabs/mobile-testkit/issues/994)